### ISSUE TITLE
Can I search a movie or an episode from a library with the file name/path of the media?

### ISSUE THREAD

**Issue by Ingvix** [pkkid/python-plexapi#362]
I looked at the documentation a bit but couldn't figure out if this can be done. I'm bit of a beginner in Python so I'd like to be sure if what I need to be done can be done with this before I really get into it. Specifically I'm after the watched status of the media and do stuff based on if it's watched or not.

**Comment by Hellowlol** [comment]
It’s possible to check if a item is watched or not. Search for isWatched in the docs.

**Comment by Ingvix** [comment]
I know that's possible. What I don't know is if I can identify the video item based on the video's actual path in the system.

**Comment by Hellowlol** [comment]
Best option would be use search then check if the file path is in iterparts

**Comment by lamyipming** [comment]
I know this is closed, but just want to leave my solution on this for anyone interested.
So basically for a file name like q="Batman.Begins.2005.2160p.4K.BluRay.x265.10bit.AAC5.1-[YTS.MX].mkv", I search for the first word "Batman", and then for each of the returned results, comprehend a list of file paths.
The for loop will loop through batman trilogy 3 movies. Or if your movie starts with common word like "The" then it possibly will have more loops, but it's still better than using movies.all()

`
import re
for movie in movies.search(re.search("\w*", q).group()):
    files = [part.file for media in movie.media for part in media.parts if part.file.find(q)>=0]
    if len(files)>0:
        movie.refresh()  #this is the movie item you are looking for, I refresh the metadata here
        break
`

**Comment by lamyipming** [comment]
I know this is closed, but just want to leave my solution on this for anyone interested.
So basically for a file name like q="Batman.Begins.2005.2160p.4K.BluRay.x265.10bit.AAC5.1-[YTS.MX].mkv", I search for the first word "Batman", and then for each of the returned results, comprehend a list of file paths.
The for loop will loop through batman trilogy 3 movies. Or if your movie starts with common word like "The" then it possibly will have more loops, but it's still better than using movies.all()

```python
import re
for movie in movies.search(re.search("\w*", q).group()):  
files = [part.file for media in movie.media for part in media.parts if part.file.find(q)>=0]  
    if len(files)>0:  
        movie.refresh()  #this is the movie item you are looking for, I refresh the metadata here  
        break  
```

**Comment by Ingvix** [comment]
Ah, I could also give the quite sketchy low-level solution I ended up with. I just simply crafted this sqlite query that returns the IDs of the users that have watched the episode in the given path. It has nothing to do with python-plexapi, but can probably done in python just as easily. I just happened to be more comfortable using bash.
```
sqlite3 \
	"/var/lib/plexmediaserver/Library/Application Support/Plex Media Server/Plug-in Support/Databases/com.plexapp.plugins.library.db" \
	"SELECT account_id
	 FROM metadata_item_views
	 INNER JOIN metadata_items ON metadata_items.guid = metadata_item_views.guid
	 INNER JOIN media_items ON media_items.metadata_item_id = metadata_items.id
	 INNER JOIN media_parts ON media_parts.media_item_id = media_items.id
	 WHERE media_parts.file = \"$1\""
``` 
Where $1 is the path of the episode you wish to check. 

I know my user's ID is 1 so I just check if the output returns a line with just that. Though it wouldn't be too hard to extend that query to provide the account name based on the ID, I just didn't see it as necessary. On Windows the database path is a bit different. Probably somewhere in %APPDATA% but inside the Plex data directory the dir tree should be quite the same.
